### PR TITLE
Update clean web cache method with new WebKit API

### PR DESCRIPTION
**Pull request by vtourraine**
- Update clean web cache method with new WebKit API
- Clean web cache at signout instead of before presenting the login view, because the new method is asynchronous
- Update library requirements to iOS 11 and macOS 10.10

### CODE REVIEW

File: MendeleyKit/MendeleyKit/Public Interface/MendeleyKit.m [hunk 1]
```diff
@@ -213,6 +213,9 @@ - (void)initialLoginStatus
 - (void)clearAuthentication
 {
     [MendeleyKitConfiguration.sharedInstance.storeProvider removeOAuthCredentials];
+#if TARGET_OS_IPHONE
```
Comment (luigitrevisi) [inline MendeleyKit/MendeleyKit/Public Interface/MendeleyKit.m]: do we still need an if here?

File: MendeleyKit/MendeleyKit/UIKit/MendeleyKitLoginHelper.swift [hunk 2]
```diff
@@ -62,23 +63,17 @@ open class MendeleyKitLoginHelper: NSObject
...
+        let dataStore = WKWebsiteDataStore.default()
+        dataStore.fetchDataRecords(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes()) { records in
+            dataStore.removeData(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes(), for: records.filter { $0.displayName.contains("mendeley.com") }) {}
         }
     }
+
+    open func cleanCookiesAndURLCache()
+    {
+        MendeleyKitLoginHelper.cleanCookiesAndURLCache()
+    }
```
Comment (luigitrevisi) [inline MendeleyKit/MendeleyKit/UIKit/MendeleyKitLoginHelper.swift]: I might be wrong: why do we need both a class and an instance method?

File: MendeleyKit/MendeleyKit/Public Interface/MendeleyKit.m (hunk 1 above)
Comment (vtourraine) [reply MendeleyKit/MendeleyKit/Public Interface/MendeleyKit.m]: Ideally, no, but right now the `MendeleyKitLoginHelper` implementation is UIKit-only, including a reference to the presenting `UIViewController`. We should refactor it to offer a fully shared implementation, but that would require more changes. For now, we still need this `#if`.

File: MendeleyKit/MendeleyKit/UIKit/MendeleyKitLoginHelper.swift (hunk 2 above)
Comment (vtourraine) [reply MendeleyKit/MendeleyKit/UIKit/MendeleyKitLoginHelper.swift]: The new implementation needs to be in a class method, but I kept the instance method for full backward compatibility. Now that you’re pointing it out, it seems overly cautious. I’ll remove the instance method, clients should be able to update the associated code very easily.

Comment (luigitrevisi) [reply MendeleyKit/MendeleyKit/UIKit/MendeleyKitLoginHelper.swift]: Awesome!

File: MendeleyKit/MendeleyKit/Public Interface/MendeleyKit.m (hunk 1 above)
Comment (luigitrevisi) [reply MendeleyKit/MendeleyKit/Public Interface/MendeleyKit.m]: Ok

**Pull request merged**